docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentWithOptions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.ML;
+using Microsoft.ML.Trainers;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.BinaryClassification
+{
+    public static class StochasticGradientDescentWithOptions
+    {
+        // In this examples we will use the adult income dataset. The goal is to predict
+        // if a person's income is above $50K or not, based on different pieces of information about that person.
+        // For more details about this dataset, please see https://archive.ics.uci.edu/ml/datasets/adult.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Download and featurize the dataset.
+            var data = SamplesUtils.DatasetUtils.LoadFeaturizedAdultDataset(mlContext);
```

Comment:
Please add examples using in-memory data structures if possible and show how to inspect those in-memory examples' predictions. If you search for "InMemory" in VS Text Explorer, you will find a few examples.

Reply:
Since these are in the API docs on the website, I actually would prefer to have the data loading part to be terse. Let's make other samples focused on the data loading aspects, and keep this spare.

---
In reply to: [259176526](https://github.com/dotnet/machinelearning/pull/2688#discussion_r259176526) [](ancestors = 259176526)

Reply:
I agree with Rogan. That would be a scenario tutorial than API sample. 

---
In reply to: [259451392](https://github.com/dotnet/machinelearning/pull/2688#discussion_r259451392) [](ancestors = 259451392,259176526)

Reply:
So this is API doc for documentation website? If yes, it makes my feeling even stronger --- ideally, to fit this example into user's own scenario, user should be able to just make minor changes. Having a text loader decreases the flexibility of this example. Thinking about scikit-learn, I don't find many of them using text file and it's super easy to start working on their modules.

Reply:
I'm not following this comment. LoadFeaturizedAdultDataset downloads the dataset and loads it into memory. It's functionally similar to sklearn datasets module which is used in many of the sklearn examples. 

>>> from sklearn import datasets
>>> iris = datasets.load_iris()
>>> digits = datasets.load_digits()

And we have SamplesUtils.DatasetUtils.LoadFeaturizedAdultDataset(). 

Users are able to copy-paste these samples and run them as-is. Please IM me if you want to further discuss this. :)



---
In reply to: [259558223](https://github.com/dotnet/machinelearning/pull/2688#discussion_r259558223) [](ancestors = 259558223)

Reply:
I'll be using this same template for all the new API samples. If you think there's a better template, please let me know soon.

---
In reply to: [259947957](https://github.com/dotnet/machinelearning/pull/2688#discussion_r259947957) [](ancestors = 259947957,259558223)